crypto/asn1/x_bignum.c

```diff
@@ -116,6 +116,12 @@ static int bn_c2i(ASN1_VALUE **pval, const unsigned char *cont, int len,
 {
     BIGNUM *bn;
 
+    /* Reject encodings that imply a negative number. */
+    if (len == 0 || (*cont & 0x80) != 0) {
+        ERR_raise(ERR_LIB_ASN1, ASN1_R_INVALID_VALUE);
+        return 0;
+    }
+
```

Comment:
There is one problem with bn_secure_c2i() that is pronounced with this change. As it calls bn_c2i but securely allocating `pval` and not freeing it when bn_c2i fails, this is an unwanted side effect. IMO bn_secure_c2i() should free the `pval` if it allocated it, in case of error.